src/Execute/ExecutionEnv.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2022 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Command} from '../Backend/Command';
+
+interface ExecutionEnv {
+  name(): string;
+  envInfo(): string;
```

Comment:
Could you add comment about what `envInfo()` should return? Comment with some example would be great.

Reply:
Um.... currently not sure, but I think this function will be used to get information about this and check certain executor run on this ExecutionEnv. I think I will add this on other PR that handle about `Executor` to check about this.

Reply:
How about introducing `envInfo()` when you really need?
ref: https://ko.wikipedia.org/wiki/YAGNI

Reply:
> It's also hard to review such function because reviewers cannot know

I absolutely agree with this. I prefer to introduce a dummy backend and ExecutionEnv to your draft. It will help both reviewers and code writer because explaining with example make things much easier. :smile: